src/contracts/GPv2Settlement.sol

```diff
@@ -133,6 +133,21 @@ contract GPv2Settlement {
     }
 
     // solhint-enable
+    modifier onlyOwner {
+        // TODO - replace with with call to AccessControl contract
+        require(true, "Only owner can call this function");
+        _;
+    }
+
+    function transferBalanceTo(
+        address[] memory tokenAddresses,
+        address receiver
+    ) public onlyOwner {
+        for (uint256 i = 0; i < tokenAddresses.length; i++) {
+            IERC20 token = IERC20(tokenAddresses[i]);
+            token.transfer(receiver, token.balanceOf(address(this)));
```

Comment:
We might want to transfer out custom amounts to:
1. keep some tokens in the contract to balance out bad price movements of Uniswap, if so desired.
2. "officially" pay out solvers for the ETH transaction fees they incurred in the settlement plus some bonus.

Reply:
Its gonna be really messy to start specifying custom amounts of various tokens... If we really want to do that I might suggest making special batch transfers in a wrapper contract of some sort after the fact. Of course we could make the function accept an array of tuples `List[Tuple[address, amount]]` or just two arrays, but then whats next, a third array of receivers? Just seems like it goes beyond expectations.

However, it might be good to give it just `(address, customAmount, receiver)` then a wrapper contract could call it as a `multiSend` with multiple different receivers and amounts over any varying amount of tokens... So, yea... I guess lets go with this!

Reply:
I agree with everything and I like the proposal!

Reply:
Nice, that makes things even cleaner!